extensions/blocks/premium-content/premium-content.php

```diff
@@ -64,6 +64,66 @@ function render_block( $attributes, $content ) {
 		return '';
 	}
 
+	if (
+		! membership_checks()
+		// Only display Stripe nudge if Upgrade nudge isn't displaying.
+		&& required_plan_checks()
+	) {
+		$stripe_nudge = render_stripe_nudge();
+		return $stripe_nudge . $content;
+	}
+
 	Jetpack_Gutenberg::load_styles_as_required( FEATURE_NAME );
 	return $content;
 }
+
+/**
+ * Server-side rendering for the stripe connection nudge.
+ *
+ * @return string Final content to render.
+ */
+function render_stripe_nudge() {
+	if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+		jetpack_require_lib( 'memberships' );
+		$blog_id  = get_current_blog_id();
+		$settings = (array) get_memberships_settings_for_site( $blog_id );
+
+		return stripe_nudge(
+			$settings['connect_url'],
+			__( 'Connect to Stripe to use this block on your site.', 'jetpack' ),
+			__( 'Connect', 'jetpack' )
+		);
+	} elseif ( jetpack_is_atomic_site() ) {
+		// On Atomic sites, the Stripe connection url is not easily available
+		// server-side, so we redirect them to the post in the editor in order
+		// to connect.
+		return stripe_nudge(
+			get_edit_post_link( get_the_ID() ),
+			__( 'Connect to Stripe to use this block on your site.', 'jetpack' ),
+			__( 'Edit post', 'jetpack' )
+		);
```

Comment:
I would love input on copy changes for the Stripe nudge on Atomic sites, to make it clearer to the user that they will be taken back to the editor.

Reply:
"Connect to Stripe in the editor to use this block on your site." ?

Reply:
Concise, +1. Thanks, copy is hard 😅